Journaley/Forms/MainForm.cs

```diff
@@ -666,25 +676,52 @@ private string RemoveLineBreaksWithinLists(string formattedString)
             for (int i = 0; i < lines.Length - 1; ++i)
             {
                 string line = lines[i];
-                string nextLine = lines[i + 1];
 
-                // Remove the "<br />" tags only if the current line and the next line are starting
-                // with the opening/closing list tags. By doing this, it can prevent removing
-                // line breaks that are intentionally added by the user.
-                if (Regex.IsMatch(line, pattern) && Regex.IsMatch(nextLine, pattern))
+                // We first incrementally check if the line begins with <p> tag. When it does,
+                // We turn on the paragraphBuilder to append the remaining lines.
+
+                // If the checks hit a line with the </p> closing, we disable check and proceed
+                // to parse the text, add <br /> tags and then adding it to the builder variable
+                // where it is assembled with the rest of the file.
+
+                // Note, this is unoptimized code and will still need work.
+                if (paragraphBegin.IsMatch(line))
+                {
+                    isNewLine = true;
+                }
+                else if (paragraphEnd.IsMatch(line))
+                {
+                    isNewLine = false;
+                }
+
+                if (isNewLine)
                 {
-                    while (line.EndsWith("<br />"))
+                    paragraphBuilder.AppendLine(line);
+
+                    if (paragraphEnd.IsMatch(line))
                     {
-                        line = line.Substring(0, line.LastIndexOf("<br />"));
+                        builder.AppendLine(line);
+                        paragraphBuilder.Clear();
+                        isNewLine = false;
                     }
                 }
+                else
+                {
+                    paragraphBuilder.AppendLine(line);
+
+                    newLineString = paragraphBuilder.ToString();
+                    newLineString = Regex.Replace(newLineString, @"^([\w\*\>\<\[][^\r\n]*)(?=\r?\n[\w\*\>\<\[].*$)", "$1<br />", RegexOptions.Multiline);
 
-                builder.AppendLine(line);
+                    builder.AppendLine(newLineString);
+                    paragraphBuilder.Clear();
+                }
             }
 
             builder.AppendLine(lines.Last());
 
-            return builder.ToString();
+            strippedOver = Regex.Replace(builder.ToString(), @"^(\r\s)", string.Empty, RegexOptions.Multiline);

```

Comment:
Did this because the output that the builder creates has a lot of very bad carriage returns, so I stripped them into empty. Tested this, did not affect the output and the HTML stayed intact.